### PR TITLE
[MID-130] added specialization and price filters to specialists page

### PR DESCRIPTION
**Pull request by dmitrenko-v**
[Ticket](https://trello.com/c/hbVrWUDA/130-%D1%8E%D0%B7%D0%B5%D1%80-%D1%84%D1%96%D0%BB%D1%8C%D1%82%D1%80%D0%B8-%D0%BF%D0%BE%D1%81%D0%B0%D0%B4%D0%B0-%D1%82%D0%B0-%D1%86%D1%96%D0%BD%D0%B0)

### CODE REVIEW

File: src/app/api/search/route.js [hunk 1]
```diff
@@ -3,6 +3,28 @@ import { NextResponse } from 'next/server';
 import { getSearchParamsFromRequest } from '@/utils/getSearchParamsFromRequest';
 import { prisma } from '@/lib/db';
 
+function getPriceFilter(prices) {
+  const pricesFilter = [];
```
Comment (alex-popov-tech) [inline src/app/api/search/route.js]: 🟠 please consider using 'map' in instead of 'forEach'

File: src/app/api/search/route.js [hunk 2]
```diff
@@ -3,6 +3,28 @@ import { NextResponse } from 'next/server';
...
+  prices.forEach(price => {
+    if (price === 'free') {
+      pricesFilter.push({ price: { equals: 0 } });
+    }
+    if (price === 'below500') {
+      pricesFilter.push({ AND: [{ price: { gt: 0 } }, { price: { lt: 500 } }] });
```
Comment (alex-popov-tech) [inline src/app/api/search/route.js]: 🟠 if i choose only '<500', will there be filter with 'AND' only?

File: src/app/api/search/route.js (hunk 1 above)
Comment (dmitrenko-v) [reply src/app/api/search/route.js]: changed this function

File: src/app/api/search/route.js (hunk 2 above)
Comment (dmitrenko-v) [reply src/app/api/search/route.js]: changed this function

**Review by Yivgha — APPROVED** [review]

**Review by WebTarantul — APPROVED** [review]

**Review by alex-popov-tech — APPROVED** [review]

**Pull request merged**